rclcpp_action/include/rclcpp_action/server.hpp

```diff
@@ -369,7 +369,11 @@ class Server : public ServerBase, public std::enable_shared_from_this<Server<Act
     if (goal_handle) {
       resp = handle_cancel_(goal_handle);
       if (CancelResponse::ACCEPT == resp) {
-        goal_handle->_cancel_goal();
+        try {
+          goal_handle->_cancel_goal();
+        } catch (const rclcpp::exceptions::RCLError & ex) {
+          return CancelResponse::REJECT;
```

Comment:
It might be nice to log a debug message here.

Reply:
:+1:  It's a good suggestion. I have pushed a commit for it.